pkg/operator/ceph/cluster/osd/update.go

```diff
@@ -128,6 +128,16 @@ func (c *updateConfig) updateExistingOSDs(errs *provisionErrors) {
 			continue
 		}
 
+		// backward compatibility for old deployments
+		if osdInfo.DeviceClass == "" {
+			deviceClassInfo, err := cephclient.OSDDeviceClasses(c.cluster.context, c.cluster.clusterInfo, []string{strconv.Itoa(osdID)})
+			if err != nil {
+				logger.Errorf("failed to get device class for existing deployment %q. %v", depName, err)
```

Comment:
The changes look good. Just to confirm, what tests were you able to run? It would be good to confirm that in the common case we don't expect those errors to be printed in the log.

Reply:
So, i had running Rook v1.6.4, then i did upgrade on image with those changes and OSD deployments after upgrade now have ROOK_OSD_DEVICE_CLASS variable set from osd crush map. Upgrade finished with `op-osd: finished running OSDs in namespace "rook-ceph"` and no errors related to changes above observed.